pkg/chain/ethereum_chain.go

```diff
@@ -27,14 +27,13 @@ type ETHChain struct {
 
 // NewRequest : Takes signed transaction data as a parameter
 // Returns a marshalled request
-func (ec *ETHChain) NewRequest(txHex string) ([]byte, error){
-	er := ethRequest {
-		ID: ec.chainID,
+func (ec *ETHChain) NewRequest(rpcURL string, txHex string) (PostRequest, error) {
+	er := ethRequest{
+		ID:      ec.chainID,
 		JSONRPC: "2.0",
-		METHOD: "eth_sendRawTransaction",
-		Params: []string{txHex},
+		METHOD:  "eth_sendRawTransaction",
+		Params:  []string{txHex},
 	}
-
 	marshalledRequest, err := json.Marshal(er)
-	return marshalledRequest, err
-}
\ No newline at end of file
+	return PostRequest{Body: marshalledRequest}, err
```

Comment:
The body needs to be sent somewhere. Add the `url` attribute to the returned `PostRequest` struct.

Reply:
https://github.com/hashcloak/Meson-plugin/pull/11/commits/183bcd0a55b79cdcecb52970ace4913fd275e659#diff-2aaf15c211b314d7c0f1cc50a2a8bbc2R42